stdlib/map.ml

```diff
@@ -356,4 +360,42 @@ module Make(Ord: OrderedType) = struct
 
     let choose = min_binding
 
+    type 'a cursor_action =
+      | Act_yield of key * 'a
+      | Act_explore of 'a t
+
+    type 'a cursor = {
+      low: key option;
+      high: key option;
+      stack: 'a cursor_action list;
+    }
```

Comment:
I believe the higher-bound is also more efficient this way (you can cut intermediate nodes). however, you are right about simplicity.